src/services/completions.ts

```diff
@@ -371,8 +370,8 @@ namespace ts.Completions {
         const newEntries = resolvingModuleSpecifiers(
             "continuePreviousIncompleteResponse",
             host,
+            codefix.createImportSpecifierResolver(file, program, host, preferences),

```

Comment:
Shouldnt this be in `IncompleteCompletionsCache` if this work was already done when completions were invoked before

Reply:
No, the incomplete completions cache is for storing completion entries between a series of requests as the user is typing additional characters of a single identifier. So with each new character, while most of the completion entries are the same, we actually have a new source file and new program, so we don't want to retain the old ones. This work is in fact not very expensive at all, but in a single request we were calling it once per auto-import instead of just once. It takes about 0.1ms, but in my example case that was happening 18k times 😄